msdk-io/msdk-io-netcdf/src/main/java/io/github/msdk/io/netcdf/NetCDFMsScan.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * (C) Copyright 2015-2016 by MSDK Development Team
+ *
+ * This software is dual-licensed under either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1 as published by the Free
+ * Software Foundation
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by the Eclipse Foundation.
+ */
+
+package io.github.msdk.io.netcdf;
+
+import java.io.IOException;
+
+import io.github.msdk.MSDKRuntimeException;
+import io.github.msdk.datamodel.impl.SimpleMsScan;
+import io.github.msdk.datamodel.msspectra.MsSpectrumType;
+import io.github.msdk.spectra.spectrumtypedetection.SpectrumTypeDetectionAlgorithm;
+import ucar.ma2.Array;
+import ucar.ma2.Index;
+import ucar.ma2.InvalidRangeException;
+import ucar.nc2.Variable;
+
+public class NetCDFMsScan extends SimpleMsScan {
+
+  private int[] scanStartPositions;
+  private float[] scanRetentionTimes;
+  private Variable massValueVariable;
+  private Variable intensityValueVariable;
+  private double massValueScaleFactor;
+  private double intensityValueScaleFactor;
+  private double[] mzValues;
+  private float[] intensityValues;
+  private Integer numOfDataPoints;
+  private MsSpectrumType spectrumType;
+
+  public NetCDFMsScan(Integer scanNumber, int[] scanStartPositions, float[] scanRetentionTimes,
+      Variable massValueVariable, Variable intensityValueVariable, double massValueScaleFactor,
+      double intensityValueScaleFactor) {
+    super(scanNumber);
+    this.scanStartPositions = scanStartPositions;
+    this.scanRetentionTimes = scanRetentionTimes;
+    this.massValueVariable = massValueVariable;
+    this.intensityValueVariable = intensityValueVariable;
+    this.massValueScaleFactor = massValueScaleFactor;
+    this.intensityValueScaleFactor = intensityValueScaleFactor;
+    this.mzValues = null;
+    this.intensityValues = null;
+    this.spectrumType = null;
+  }
+
+  @Override
+  public float[] getIntensityValues(float[] intensityValues) {
```

Comment:
If the Predicate was set for the scan, this function should return the pre-loaded data